### PR TITLE
[1/N] Quantization refactor: split AutoRound into package layout

### PR DESCRIPTION
**Pull request by LProDreamAll**
## Motivation

Split `AutoRound` quantization module from a single-file module (`quantization/auto_round.py`) into a package layout for cleaner code organization. The current `auto_round.py` is a monolithic module that mixes config definition, per-layer config resolution, and AWQ/GPTQ dispatch logic.

## Modifications

**Directory structure change:**

- Replaced `python/sglang/srt/layers/quantization/auto_round.py` with package `python/sglang/srt/layers/quantization/auto_round/`, containing:
  - `auto_round/auto_round.py` — preserves `AutoRoundConfig` class, public API unchanged.
  - `auto_round/routing.py` — moves `get_layer_config`, `apply_awq_quant_layer`, `apply_gptq_quant_layer` routing logic here.

- `AutoRoundConfig.get_quant_method` now delegates to the `routing` module.

**Public import path remains unchanged:**

```python
from sglang.srt.layers.quantization.auto_round import AutoRoundConfig
```

**Pre-existing bugs on `main` fixed alongside:**

1. **GPTQ + `FusedMoE` routing logic was inverted**: `use_marlin=True` incorrectly used `MoeWNA16Config`, while `use_marlin=False` could hit `NameError`. Fixed to match the AWQ path logic (`GPTQMarlinMoEMethod` vs `MoeWNA16Config`).

2. **`backend` parameter was not passed to routing functions**: `get_quant_method` called routing functions without passing `backend`, causing the configured backend to be ignored and always fall back to `"auto"`. Fixed by explicitly passing `backend=self.backend`.

## Out of scope

No dedicated `auto_round/schemes/` package. AutoRound continues to delegate to existing AWQ/GPTQ/Marlin/MoE-WNA16 implementations.

## Testing

- No new unit tests; covered by existing AutoRound CI/manual tests.
- No performance impact — no kernel changes.

### CODE REVIEW

**Comment by LProDreamAll** [comment]
@ping1jing2

**Comment by LProDreamAll** [comment]
@BBuf @ch-wan Could you please help trigger CI and review this AutoRound quantization refactor? CI needs the run-ci label. Thanks!

**Comment by LProDreamAll** [comment]
@TamirBaydasov  Can you help me?

**Comment by TamirBaydasov** [comment]
> @TamirBaydasov Can you help me?

https://github.com/sgl-project/sglang/pull/26942#issuecomment-4616632300

**Comment by TamirBaydasov** [comment]
pls, address gemini comments. Other than that, LGTM

File: python/sglang/srt/layers/quantization/auto_round/routing.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+                "quant_method": "gptq",
+                "bits": weight_bits,
+                "group_size": group_size,
+                "sym": sym,
+                "lm_head": False,
+            }
+            return MoeWNA16Config.from_config(moe_config).get_quant_method(
+                layer, prefix
+            )
+        return GPTQMarlinMoEMethod(quant_args_marlin)
```
[the remark this answers is not in the document]
Comment (LProDreamAll) [reply python/sglang/srt/layers/quantization/auto_round/routing.py]: Fixed in [994724c](https://github.com/sgl-project/sglang/pull/26846/commits/994724c), thanks!

File: python/sglang/srt/layers/quantization/auto_round/auto_round.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        """
+        raise NotImplementedError
+
+    def get_quant_method(self, layer: torch.nn.Module, prefix: str):
+        # TODO enable CPU quant method later
+        if "gptq" in self.packing_format or "gptq" in self.backend:
+            return routing.apply_gptq_quant_layer(self, layer, prefix)
+        if "awq" in self.packing_format or "awq" in self.backend:
+            return routing.apply_awq_quant_layer(self, layer, prefix)
+        return None
```
[the remark this answers is not in the document]
Comment (LProDreamAll) [reply python/sglang/srt/layers/quantization/auto_round/auto_round.py:132]: Fixed in [994724c](https://github.com/sgl-project/sglang/pull/26846/commits/994724c), thanks!

**Comment by LProDreamAll** [comment]
@TamirBaydasov  Already addressed in [994724c](https://github.com/sgl-project/sglang/pull/26846/commits/994724c). The Gemini comments (backend parameter passing and GPTQ MoE routing logic) were fixed in that commit.
